rpc_storage_handler.go

```diff
@@ -145,22 +147,41 @@ func (r *RPCStorageHandler) Connect() bool {
 	// RPC Client is unset
 	// Set up the cache
 	log.Info("Setting new RPC connection!")
-	if config.SlaveOptions.UseSSL {
-		clientCfg := &tls.Config{
-			InsecureSkipVerify: config.SlaveOptions.SSLInsecureSkipVerify,
-		}
 
-		RPCCLientSingleton = gorpc.NewTLSClient(r.Address, clientCfg)
-	} else {
-		RPCCLientSingleton = gorpc.NewTCPClient(r.Address)
-	}
+	RPCCLientSingleton = &gorpc.Client{Addr: r.Address}
+	RPCClientSingletonConnectionID = uuid.NewV4().String()
 
 	if log.Level != logrus.DebugLevel {
 		gorpc.SetErrorLogger(gorpc.NilErrorLogger)
 	}
 
 	RPCCLientSingleton.OnConnect = r.OnConnectFunc
 	RPCCLientSingleton.Conns = 50
+	RPCCLientSingleton.Dial = func(addr string) (conn io.ReadWriteCloser, err error) {
+		dialer := &net.Dialer{
+			Timeout:   10 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}
+
+		if config.SlaveOptions.UseSSL {
+			cfg := &tls.Config{
+				InsecureSkipVerify: config.SlaveOptions.SSLInsecureSkipVerify,
+			}
+
+			conn, err = tls.DialWithDialer(dialer, "tcp", addr, cfg)
+		} else {
+			conn, err = dialer.Dial("tcp", addr)
+		}
+
+		if err != nil {
+			return
+		}
+
+		conn.Write([]byte("register"))
+		conn.Write([]byte{byte(len(RPCClientSingletonConnectionID))})
+		conn.Write([]byte(RPCClientSingletonConnectionID))
```

Comment:
> the first two are so that we can change this "protocol" in the future more easily.
Exactly

> The silent overflow is a bit scary, even if we're sure that we won't touch this very often.
Good point, because if we would like to change protocol, we for sure forgot about this limitation.